snakemake/workflow.py

```diff
@@ -798,15 +824,17 @@ def generate_unit_tests(self, path: Path):
         self._build_dag()
 
         deploy = []
-        if DeploymentMethod.CONDA in self.deployment_settings.deployment_method:
-            deploy.append("conda")
-        if DeploymentMethod.APPTAINER in self.deployment_settings.deployment_method:
-            deploy.append("singularity")
+        if self.deployment_settings is not None:
```

Comment:
why would this be none here?

Reply:
These following checks of None are added as these params are setted as None when it was created. `mypy` warned me that. Probably an assert is proper here, or I will ignore these type of warnings? Thanks for your advice!